### PR TITLE
[FEAT] Add Koalabear field support

### PR DESCRIPTION
**Pull request by nonam3e**
## Describe the changes

This PR adds koalabear field into icicle

| Task                      | Status   |
|---------------------------|----------|
| Koalabear support          | ✅ Done  |
| Extension field support    | ✅ Done  |
| Rust bindings              | ✅ Done  |
| Golang bindings            | ✅ Done  |
| Poseidon support           | ✅ Done  |
| Poseidon2 support          | ✅ Done  |
| CI         | ✅ Done  |

### CODE REVIEW

File: icicle/include/icicle/api/koalabear.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// WARNING: This file is auto-generated by a script.
+// Any changes made to this file may be overwritten.
+// Please modify the code generation script instead.
+// Path to the code generation script: scripts/gen_c_api.py
+
+#pragma once
```
Comment (yshekel) [inline icicle/include/icicle/api/koalabear.h:6]: @ChickenLover @nonam3e do we even need those files? I suggest removing them. it's probably never used and no reason to use them.

**Review by yshekel — APPROVED** [review]

File: icicle/include/icicle/api/koalabear.h (hunk 1 above)
Comment (ChickenLover) [reply icicle/include/icicle/api/koalabear.h:6]: agree

File: icicle/include/icicle/hash/poseidon_constants/constants/rounds_numbers_coalabear.txt
Comment (jeremyfelder) [inline icicle/include/icicle/hash/poseidon_constants/constants/rounds_numbers_coalabear.txt:1]: Is this a temp file?

Comment (nonam3e) [reply icicle/include/icicle/hash/poseidon_constants/constants/rounds_numbers_coalabear.txt:1]: yeah, deleted

File: icicle/include/icicle/api/koalabear.h (hunk 1 above)
Comment (nonam3e) [reply icicle/include/icicle/api/koalabear.h]: ok, deleted them

Comment (nonam3e) [reply icicle/include/icicle/api/koalabear.h:6]: sorry, didn't noticed that current examples depend on these files. It's better to remove them in another pr

Comment (LeonHibnik) [reply icicle/include/icicle/api/koalabear.h:6]: @yshekel we need to discuss about it removing them, and in another PR ofc

**Review by LeonHibnik — APPROVED** [review]

**Comment by jeremyfelder** [comment]
We should update the docs to include the new curve as well

**Review by jeremyfelder — APPROVED** [review]

**Review by LeonHibnik — APPROVED** [review]

**Pull request merged**